pkg/endpoint/policy.go

```diff
@@ -709,3 +710,48 @@ func (e *Endpoint) GetCIDRPrefixLengths() (s6, s4 []int) {
 	}
 	return e.desiredPolicy.CIDRPolicy.ToBPFData()
 }
+
+// UpdateVisibilityPolicy updates the visibility policy of this endpoint to
+// reflect the state stored in the proxy visibility annotation, if present,
+// in anno. If no such annotation is present, then the VisibilityPolicy for the
+// Endpoint will be empty, and will have no effect. If the proxy visibility
+// annotation cannot be parsed, an empty visibility policy is assigned to the
+// Endpoint.
+func (e *Endpoint) UpdateVisibilityPolicy(anno map[string]string) {
+	if err := e.lockAlive(); err != nil {
+		// If the endpoint is being deleted, we don't need to update its
+		// visibility policy.
+		return
+	}
+
+	defer func() {
+		// Ensure that policy computation is performed so that endpoint
+		// desiredPolicy and realizedPolicy pointers are different. This state
+		// is needed to update endpoint policy maps with the policy map state
+		// generated from the visibility policy. This can, and should be more
+		// elegant in the future.
+		e.forcePolicyComputation()
```

Comment:
Should this only be done at the end of the function (ie in the success case)? If `NewVisibilityPolicy()` fails, this will not only fail to apply visibility policy, but also unnecessarily force computation.

Reply:
Note that below we disable visibility policy if an invalid visibility policy is imported. So, we are always changing the visibility policy in either success or failure cases. In order to realize this state in either case we have to force computation of policy.

Reply:
OK, thanks! I didn't glean this from the comment here, if you think it's useful to leave this context in the code then maybe mention that as well.